core/src/utils/internal/dom.ts

```diff
@@ -95,3 +99,136 @@ let idCount = 0;
  * @returns The generated ID.
  */
 export const generateId = () => `auId-${idCount++}`;
+const notEmpty = (value: any) => value != null && value !== false;
+
+/**
+ * Creates a virtual node used for server-side rendering (SSR).
+ * This function generates a mock HTML element with basic functionality,
+ * such as adding and removing event listeners, setting and removing attributes,
+ * and managing styles.
+ *
+ * @returns A virtual HTML element with simulated DOM manipulation functions.
+ */
+export function virtualNode() {
+	const attributes: Record<string, string | undefined> = {};
+	const style: Record<string, string | undefined> = {};
+
+	const classNames = new Set<string>();
+	const classListApi = {
+		add: (className: string) => classListApi.toggle(className, true),
+		remove: (className: string) => classListApi.toggle(className, false),
+		toggle(className: string, addClass: boolean) {
+			if (addClass) {
+				classNames.add(className);
+			} else {
+				classNames.delete(className);
+			}
+		},
+	};
+
+	const node = {
+		addEventListener: noop,
+		removeEventListener: noop,
+		setAttribute(name: string, value: string) {
+			attributes[name] = value;
```

Comment:
I think setting the `class` and `style` attributes should work in the virtual node just like in the browser.